src/sentry/stacktraces/processing.py

```diff
@@ -192,16 +195,15 @@ def find_stacktraces_in_data(
     rv = []
 
     def _append_stacktrace(stacktrace: Any, container: Any, is_exception: bool = False) -> None:
-        if not is_exception and (not stacktrace or not get_path(stacktrace, "frames", filter=True)):
+        frames = get_path(stacktrace, "frames", filter=True, default=())
+        if not is_exception and (not stacktrace or not frames):
             return
 
-        platforms = {
-            frame.get("platform") or data.get("platform")
-            for frame in get_path(stacktrace, "frames", filter=True, default=())
-        }
+        platforms = _get_frames_metadata(frames, data.get("platform", "unknown"))
         rv.append(
             StacktraceInfo(
                 stacktrace=stacktrace,
+                # XXX: This contains all the contents of stacktrace
                 container=container,
```

Comment:
In the future, we should extract the frames directly from the container rather than from stacktrace or even determine why we need a container at all.

Reply:
There are certain platforms where we need the container even if there's nothing in it (or nothing that's not `None` or empty). See https://github.com/getsentry/sentry/pull/15795#discussion_r349916876.